3rdparty/hal_rvv/hal_rvv_1p0/transpose.hpp

```diff
@@ -0,0 +1,220 @@
+        int *dst = (int*)(dst_data) + h;
+        int vl;
+        for (w = 0; w < src_width; w += vl) {
+            vl = __riscv_vsetvl_e32m8(src_width - w);
+            auto v = __riscv_vle32_v_i32m8(src + w, vl);
+            __riscv_vsse32(dst + w * dst_step_base, dst_step, v, vl);
+        }
+    }
+}
+
+static void transpose2d_32sC2(const uchar *src_data, size_t src_step, uchar *dst_data, size_t dst_step, int src_width, int src_height) {
+    auto transpose_64s_8xVl = [](const int64_t *src, size_t src_step, int64_t *dst, size_t dst_step, const int vl) {
+        auto v0 = __riscv_vle64_v_i64m1(src, vl);
+        auto v1 = __riscv_vle64_v_i64m1(src + src_step, vl);
+        auto v2 = __riscv_vle64_v_i64m1(src + 2 * src_step, vl);
+        auto v3 = __riscv_vle64_v_i64m1(src + 3 * src_step, vl);
+        auto v4 = __riscv_vle64_v_i64m1(src + 4 * src_step, vl);
+        auto v5 = __riscv_vle64_v_i64m1(src + 5 * src_step, vl);
+        auto v6 = __riscv_vle64_v_i64m1(src + 6 * src_step, vl);
+        auto v7 = __riscv_vle64_v_i64m1(src + 7 * src_step, vl);
+        vint64m1x8_t v = __riscv_vcreate_v_i64m1x8(v0, v1, v2, v3, v4, v5, v6, v7);
+        __riscv_vssseg8e64(dst, dst_step, v, vl);
+    };
+
+    size_t src_step_base = src_step / sizeof(int64_t);
+    size_t dst_step_base = dst_step / sizeof(int64_t);
+
+    int h = 0, w = 0;
+    for (; h <= src_height - 8; h += 8) {
+        const int64_t *src = (const int64_t*)(src_data) + h * src_step_base;
+        int64_t *dst = (int64_t*)(dst_data) + h;
+        int vl;
+        for (w = 0; w < src_width; w += vl) {
+            vl = __riscv_vsetvl_e64m1(src_width - w);
+            transpose_64s_8xVl(src + w, src_step_base, dst + w * dst_step_base, dst_step, vl);
+        }
+    }
+    for (; h < src_height; h++) {
+        const int64_t *src = (const int64_t*)(src_data) + h * src_step_base;
+        int64_t *dst = (int64_t*)(dst_data) + h;
+        int vl;
+        for (w = 0; w < src_width; w += vl) {
+            vl = __riscv_vsetvl_e64m8(src_width - w);
+            auto v = __riscv_vle64_v_i64m8(src + w, vl);
+            __riscv_vsse64(dst + w * dst_step_base, dst_step, v, vl);
+        }
+    }
+}
+
+#undef cv_hal_transpose2d
+#define cv_hal_transpose2d cv::cv_hal_rvv::transpose::transpose2d
+
+using Transpose2dFunc = void (*)(const uchar*, size_t, uchar*, size_t, int, int);
+inline int transpose2d(const uchar* src_data, size_t src_step, uchar* dst_data, size_t dst_step,
+                       int src_width, int src_height, int element_size) {
+    if (src_data == dst_data) {
+        return CV_HAL_ERROR_NOT_IMPLEMENTED;
+    }
+
+    static Transpose2dFunc tab[] = {
+        0, transpose2d_8u, transpose2d_16u, 0,
+        transpose2d_32s, 0, 0, 0,
+        transpose2d_32sC2, 0, 0, 0,
+        0, 0, 0, 0,
+        0, 0, 0, 0,
+        0, 0, 0, 0,
+        0, 0, 0, 0,
+        0, 0, 0, 0,
+        0
+    };
+    Transpose2dFunc func = tab[element_size];
```

Comment:
`element_size` may trigger out-of-bound access issue. I propose to add range check before.

Reply:
It is checked before calling hal:

https://github.com/opencv/opencv/blob/e37819c2ac1b8ec2beac917bba0d885c79195603/modules/core/src/matrix_transform.cpp#L181